File: src/org/elixir_lang/heex/File.java
package org.elixir_lang.heex;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import org.elixir_lang.heex.Language;
import org.elixir_lang.heex.file.Type;
import org.jetbrains.annotations.NotNull;

public class File extends PsiFileBase {
    public File(@NotNull FileViewProvider fileViewProvider) {
        super(fileViewProvider, Language.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return Type.INSTANCE;
    }

    @NotNull
    @Override
    public String toString() {
        return "HTML Embedded Elixir File";
    }
}

File: src/org/elixir_lang/beam/assembly/File.kt
package org.elixir_lang.beam.assembly

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider
import org.elixir_lang.icons.ElixirIcons
import javax.swing.Icon

class File(fileViewProvider: FileViewProvider): PsiFileBase(fileViewProvider, Language) {
    override fun getFileType(): FileType = org.elixir_lang.beam.assembly.file.Type
    override fun getIcon(flags: Int): Icon = ElixirIcons.FILE
}
